tensorflow/python/eager/backprop.py

```diff
@@ -951,6 +955,11 @@ def gradient(self,
     if output_gradients is not None:
       output_gradients = [None if x is None else ops.convert_to_tensor(x)
                           for x in nest.flatten(output_gradients)]
+    
+    for v in nest.flatten(target):
```

Comment:
I want you to avoid computing nest.flatten(target) twice; so iterate over flat_targets